provider.go

```diff
@@ -458,6 +461,17 @@ func (p *Provider) MountSecretsStoreObjectContent(ctx context.Context, attrib ma
 	return nil
 }
 
+func validateFilePath(path string) error {
+	segments := strings.Split(strings.ReplaceAll(path, `\`, "/"), "/")
```

Comment:
Do we need to split the path by segments? Since `..` has a special meaning on paths, we should be fine if we check the string as a whole to see if that's present, similar to the check in [here](https://github.com/hashicorp/vault/blob/master/sdk/logical/storage_view.go#L100).

Reply:
How so? If I do `mkdir foo..bar` on my machine, that creates a single folder called `foo..bar` which I can interact with normally.

However, regardless of that, I am definitely ok with a simple `string.Contains` check for `..` as that's far harder to cheat, and still seems like a very reasonable limitation.

Reply:
You're right, `..` can be part of a filename.